docs/src/main/sphinx/connector/hudi.rst

```diff
@@ -0,0 +1,152 @@
+==============
+Hudi connector
+==============
+
+.. raw:: html
+
+  <img src="../_static/img/hudi.png" class="connector-logo">
+
+The Hudi connector enables querying `Hudi <https://hudi.apache.org/docs/overview/>`_
+tables synced to Hive metastore. The connector uses the metastore only to track
+partition locations. It makes use of the underlying Hudi filesystem and input formats
+to list data files. To learn more about the design of the connector, please check out
+`RFC-40 <https://github.com/apache/hudi/blob/master/rfc/rfc-40/rfc-40.md>`_.
+
+Requirements
+------------
+
+To use Hudi, we need:
+
+* Network access from the Trino coordinator and workers to the distributed object storage.
+* Access to a Hive metastore service (HMS).
+* Network access from the Trino coordinator to the HMS.
+
+Configuration
+-------------
+
+The connector requires a Hive metastore for table metadata and supports the same
+metastore configuration properties as the :doc:`Hive connector
+</connector/hive>`. At a minimum, ``hive.metastore.uri`` must be configured.
+
+The connector recognizes Hudi tables synced to the metastore by the Hudi sync tool.
+If non-Hudi tables are present in the metastore, as well, they are not visible to
+the connector.
+
+To configure the Hudi connector, create a catalog properties file, for
+example ``etc/catalog/hudi.properties``, that references the ``hudi``
+connector. Update the ``hive.metastore.uri`` with the URI of your Hive metastore
+Thrift service:
+
+.. code-block:: properties
+
+    connector.name=hudi
+    hive.metastore.uri=thrift://example.net:9083
+
+Additionally, following configuration properties can be set depending on the use-case.
+
+.. list-table:: Hudi configuration properties
+    :widths: 30, 55, 15
+    :header-rows: 1
+
+    * - Property name
+      - Description
+      - Default
+    * - ``hudi.metadata-enabled``
+      - Fetch the list of file names and sizes from metadata rather than storage.
+      - ``false``
+    * - ``hudi.columns-to-hide``
+      - List of column names that will be hidden from the query output.
+        It can be used to hide Hudi meta fields. By default, no fields are hidden.
+      -
+    * - ``hudi.use-metastore-for-partitions``
+      - Fetch partition info from the metastore. If disabled, then the partition
```

Comment:
Please do add more details about this setting.

See also https://github.com/trinodb/trino/pull/10228/files#r912650933